### PR TITLE
C++: Support callback-based generic service

### PR DESCRIPTION
**Pull request by vjpai**
Allow the use of callback API for generic services.

### CODE REVIEW

**Comment by vjpai** [comment]
This popped up an msan bug that is not related (it's on non-generic API) but it's the first time I've seen it, so I went ahead and added it in 397bdd6

File: src/cpp/server/server_cc.cc [hunk 1]
```diff
@@ -348,17 +349,28 @@ class Server::SyncRequest final : public internal::CompletionQueueTag {
...
   CallbackRequest(Server* server, size_t method_idx,
                   internal::RpcServiceMethod* method, void* method_tag)
       : server_(server),
         method_index_(method_idx),
         method_(method),
         method_tag_(method_tag),
         has_request_payload_(
-            method->method_type() == internal::RpcMethod::NORMAL_RPC ||
-            method->method_type() == internal::RpcMethod::SERVER_STREAMING),
+            method_ != nullptr &&
```
Comment (yashykt) [inline src/cpp/server/server_cc.cc]: add comment that method_ != nullptr is for generic services

**Comment by vjpai** [comment]
Thanks for the review and approval. I added a commit to address your comments, and then squashed.

**Comment by vjpai** [comment]
Perfectly green!

**Comment by vjpai** [comment]
Perfectly green!

**Pull request merged**